packages/side-nav/src/SideNavItem.tsx

```diff
@@ -232,25 +207,13 @@ const SideNavItem = React.forwardRef((props: SideNavItemProps, forwardRef) => {
 
   const { usingKeyboard: showFocus } = useUsingKeyboardContext();
 
-  let Root: ElementType<any>;
-  let role: string | null;
-
-  if (usesLinkElement(props)) {
-    Root = RootComponentTypes.Link;
-    role = props.role || 'menuitem';
-  } else if (usesCustomElement(props)) {
-    Root = props[RootComponentTypes.Custom];
-    role = props.role || null;
-  } else {
-    Root = RootComponentTypes.Button;
-    role = props.role || 'menuitem';
-  }
-
   return (
     <li role="none">
-      <Root
+      <Box
+        as='button'
         {...rest}
         {...sideNavItemContainer.prop}
+        role='menuitem'
```

Comment:
shouldn't this be before the rest so that it can be overridden?

Reply:
yes good catch